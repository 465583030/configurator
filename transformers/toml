#!/usr/bin/ruby
require "rubygems"
require "json"

## An incomplete but totally usable transformer for TOML
## TODO: nested tables, date support?

def jsonize(obj)
	# Ruby JSON is picky about producing JSON for literals
	JSON.dump([obj])[1...-1]
end

def transform(obj)
	if obj.is_a? Hash
		obj.collect do |key,value| 
			if value.is_a? Hash
				"[#{key}]\n#{transform(value)}\n"
			else
				"#{key} = #{transform(value)}"
			end
		end .join "\n"
	else
		jsonize(obj)
	end
end

puts transform(JSON.parse(STDIN.read))