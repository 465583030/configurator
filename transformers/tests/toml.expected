title = "TOML Example"
[owner]
name = "Tom Preston-Werner"
organization = "Github"
bio = "GitHub Cofounder & CEO\nLikes tater tots and beer."
dob = "2012-04-23T18:25:43.511Z"

[database]
server = "192.168.1.1"
ports = [8001,8001,8002]
connection_max = 5000
enabled = true

[clients]
data = [["gamma","delta"],[1,2]]

hosts = ["alpha","omega"]
